lighthouse-core/test/lib/timing-trace-saver-test.js

```diff
@@ -37,92 +36,11 @@ const mockEntries = [{
   entryType: 'measure',
 },
 ];
-const expectedTrace = {
-  traceEvents: [{
-    name: 'lh:init:config',
-    cat: 'measure',
-    ts: 650000,
-    dur: 210000,
-    args: {},
-    pid: 0,
-    tid: 50,
-    ph: 'X',
-    id: '0x0',
-  },
-  {
-    name: 'lh:runner:run',
-    cat: 'measure',
-    ts: 870000,
-    dur: 120000,
-    args: {},
-    pid: 0,
-    tid: 50,
-    ph: 'X',
-    id: '0x1',
-  },
-  {
-    name: 'lh:runner:auditing',
-    cat: 'measure',
-    ts: 990000,
-    dur: 750000,
-    args: {},
-    pid: 0,
-    tid: 50,
-    ph: 'X',
-    id: '0x2',
-  },
-  {
-    name: 'lh:audit:is-on-https',
-    cat: 'measure',
-    ts: 1010000,
-    dur: 10000,
-    args: {},
-    pid: 0,
-    tid: 50,
-    ph: 'X',
-    id: '0x3',
-  },
-  ],
-};
-
 
 describe('generateTraceEvents', () => {
-  let consoleError;
-  let origConsoleError;
-
-  beforeEach(() => {
-    origConsoleError = console.error;
-    consoleError = jest.fn();
-    console.error = consoleError;
-  });
-
-  afterEach(() => {
-    console.error = origConsoleError;
-  });
-
-  it('generates a single trace event', () => {
-    const event = generateTraceEvents(mockEntries);
-    assert.deepStrictEqual(event.slice(0, 1), expectedTrace.traceEvents.slice(0, 1));
-  });
-
-  it('doesn\'t allow overlapping events', () => {
-    const overlappingEntries = [{
-      startTime: 10,
-      name: 'overlap1',
-      duration: 100,
-      entryType: 'measure',
-    },
-    {
-      startTime: 30,
-      name: 'overlap2',
-      duration: 100,
-      entryType: 'measure',
-    },
-    ];
-
-    generateTraceEvents(overlappingEntries);
-    expect(consoleError).toHaveBeenCalled();
-    expect(consoleError.mock.calls[0][0]).toContain('measures overlap');
+  it('generates a pair of trace events', () => {
+    const events = generateTraceEvents([mockEntries[0]]);
+    expect(events.slice(0, 1)).toMatchSnapshot();
   });
 });
```

Comment:
Add a test to hit a zero duration event?

```javascript
  it('Handles zero duration events', () => {
    const overlappingEntries = [
      {
        startTime: 10,
        name: 'zeroLength',
        duration: 0,
        entryType: 'measure',
      },
    ];
    const expected = {
      name: 'zeroLength',
      cat: 'measure',
      ts: 10000,
      dur: 0,
      args: {},
      pid: 0,
      tid: 50,
      ph: 'n',
      s: 't',
      id: '0x0',
    };
    const event = generateTraceEvents(overlappingEntries);
    assert.deepStrictEqual(event[0], expected);
  });
```

Reply:
good call. actually we dont need the `ph:'n'` now that we're on async events. 
problem solved. :)